src/message-handler/src/MessageDispatcher.cpp

```diff
@@ -0,0 +1,98 @@
+#include "MessageDispatcher.h"
+#include <bsp/SettingsContainer.h>
+
+MessageDispatcher::MessageDispatcher(ICircularQueue<MessageDTO>& buzzOutputQ,
+                                     ICircularQueue<MessageDTO>& hostOutputQ,
+                                     ICircularQueue<MessageDTO>& remoteOutputQ,
+                                     IHiveMindHostDeserializer& deserializer,
+                                     uint16_t bspUuid,
+                                     ILogger& logger) :
+    m_buzzOutputQueue(buzzOutputQ),
+    m_hostOutputQueue(hostOutputQ),
+    m_remoteOutputQueue(remoteOutputQ),
+    m_deserializer(deserializer),
+    m_bspUuid(bspUuid),
+    m_logger(logger) {}
+
+bool MessageDispatcher::deserializeAndDispatch() {
+    MessageDTO message;
+    if (m_deserializer.deserializeFromStream(message)) {
+
+        uint32_t destinationId = message.getDestinationId();
+        if (destinationId == m_bspUuid) {
+            return dispatchMessage(message);
+        }
+
+        // Message is for a remote host
+        m_logger.log(LogLevel::Debug, "Message sent to remote queue");
+        m_remoteOutputQueue.push(message);
+        return true;
+    }
+
+    // Could not find the destination
+    m_logger.log(LogLevel::Warn, "Destination could not be found");
```

Comment:
If we end up here, it's because `deserializeFromStream()` failed, not because destination could not be found